src/java.base/share/classes/java/io/BufferedReader.java

```diff
@@ -394,12 +395,48 @@ String readLine(boolean ignoreLF, boolean[] term) throws IOException {
      *
      * @throws     IOException  If an I/O error occurs
      *
+     * @see #readAllLines
      * @see java.nio.file.Files#readAllLines
      */
     public String readLine() throws IOException {
         return readLine(false, null);
     }
 
+    /**
+     * Reads all remaining lines of text. A line is considered to be terminated
+     * by any one of a line feed ('\n'), a carriage return ('\r'), a carriage
+     * return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
+     * <p> This method works as if invoking it were equivalent to evaluating
+     * the expression:
+     * <blockquote>{@linkplain #lines()}.toList()</blockquote>
```

Comment:
I'd suggest avoiding being too specific here, as this implies that this method will invoke the `lines()` method which can be overridden by a subclass. One could have an argument about whether this indeed is implied by the "as if" wording but I think it would be preferable to avoid this. I also think we don't want to have this method depend on possible overridden methods, so that we can provide a future optimized implementation without breaking something. Thus I think most of this doc comment can be inherited form Reader, even though the implementation might be overridden.